Robot2018/src/org/usfirst/frc/team199/Robot2018/subsystems/Lift.java

```diff
@@ -2,39 +2,80 @@
 
 import org.usfirst.frc.team199.Robot2018.Robot;
 import org.usfirst.frc.team199.Robot2018.RobotMap;
-import org.usfirst.frc.team199.Robot2018.commands.AutoLift;
+import org.usfirst.frc.team199.Robot2018.commands.UpdateLiftPosition;
 
 import edu.wpi.first.wpilibj.Encoder;
 import edu.wpi.first.wpilibj.SpeedControllerGroup;
-import edu.wpi.first.wpilibj.command.Command;
-import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj.command.PIDSubsystem;
 
 /**
  *
  */
-public class Lift extends Subsystem implements LiftInterface {
+public class Lift extends PIDSubsystem implements LiftInterface {
 
 	private final SpeedControllerGroup liftMotors = RobotMap.liftMotors;
 	private final Encoder liftEnc = RobotMap.liftEnc;
-	private Position currPosition = Position.GROUND;
+	private LiftHeight currPosition = LiftHeight.GROUND;
+
+	private final int NUM_STAGES;
+	private final double WIGGLE_ROOM;
+	private final double SWITCH_DIST;
+	private final double SCALE_DIST;
+	private final double BAR_DIST;
+
+	public Lift() {
+		super("Lift", Robot.getConst("LiftkP", 0.1), Robot.getConst("LiftkI", 0), Robot.getConst("LiftkD", 0),
+				Robot.getConst("LiftkF", 0.1));
```

Comment:
We should be able to calculate reasonable defaults for the above constants to minimize or eliminate tuning.